### ISSUE TITLE
Reuse policy output

### ISSUE THREAD

**Issue by mboret** [cloud-custodian/cloud-custodian#5473]
Hi,

I wondering if it's possible to reuse policy output from another policy. My use case is, we have a lot of EKS cluster created for test purpose. Sometimes, people think to delete cluster nodes but forget to delete the EKS cluster. So I need to identify these EKS clusters. The only way(IMO), will be to filter ec2 instance by tag "eks:cluster-name" and check if for each EKS cluster I have at least one EC2 running instance.

Another use case, I would like to get the size of EBS volumes attached to stopped instances. This is not possible with a unique policy as EC2 describe gives only the volumeID. So this needs to be in two steps, get the volumeID list of stopped instances(first policy -ec2-) and then describe ebs volumes(second policy -ebs-). 

Thanks,

**Comment by kapilt** [comment]
its possible via value_from:  .. its also possible to just implement additional filters, ie ec2 has an ebs filter and vice versa to support this sort of use case. ie you can pull and filter on related resources, albeit there needs to be a filter implementation for the same, and it sounds like your also looking for some set logic (like a value_type:sum perhaps), it feels like both of those uses cases would need some contribution to enable. re the eks cluster are the node manually stood up ec2 or an asg?

**Comment by mboret** [comment]
Thanks for your answer @kapilt .

 So, if I want EBS volumes attached to stopped instances:

```
policies:
    - name: ebs-attached-to-stopped-instances
      resource: ebs
      filters:
        - type: instance
          key: State.Name
          value: stopped
```
But I can't get only those attached to stopped instances for more than 15 days. Correct? (As I cannot use the EC2 filter "state-age"). So, for this advanced use case, I should use value_from but I don't know how to "dynamically"(path can change regarding the AWS region if I pass multiple time the "--region" flag) find the resources.json file's path of my EC2 stopped instances.
It should be something like this? (not working)

```
policies:
  - name: ec2-instance-stopped-for-more-than-15-days
    resource: ec2
    filters:
      - State.Name: stopped
      - type: state-age
        op: greater-than
        days: 15
    - name: ebs-attached-to-stopped-instances
      resource: ebs
      filters:
       - type: value
         key: "VolumeId"
         op: in
         value_from:
             url: file:////output/{aws_region}/ec2-instance-stopped-for-more-than-15-days/resources.json
             format: json
             expr: '[].BlockDeviceMappings[].Ebs.VolumeId'
```

For, EKS, nodes are managed by an AutoScaling Group.

Thanks

**Comment by mboret** [comment]
Ok, got it. By replacing '{aws_region}' by '{{aws_region}' it works. Last point, is it possible to get the OUTPUT_DIR Cli parameter as a variable? To be able to build a more flexible solution, because in my case, if the OUTPUT_DIR parameter is not equal to "output", it won't work.

**Comment by kapilt** [comment]
output dir is technically customizable, it can be specified as with variables {account_id, region, policy, now) etc to control location. ie if we add in the policy's current output directory as a variable, there's no clear de-referencing to another policy's location, outside of knowledge of the execution. effectively this sort of consumption is fragile/always dependent on operator/policy author knowledge of the execution flags/path. i was just looking at how that custom path interacts with region suffixing on multiple regions, it looks like the later does not respect it, i'll file a separate issue on that.

we did have a pr, that allowed for related resource filters to pull in additional arbitrary resource filters, which might be worth revisiting as additional way of expression, part of the concern at the time was that policy authors could get themselves in trouble with the additional api calls, but we've surfaced api call visibility for policies much more (api stats, and tracing) such that even if we give rope to hang, we also give tools to cut the rope ;-) enough to compensate.
